flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -622,6 +627,43 @@ private Map<StreamConfig.SourceInputConfig, ChainedSource> createChainedSources(
         return chainedSourceInputs;
     }
 
+    // Get the numRecordsOut counter for the operator represented by the given config. And re-use
+    // this operator-level counter for the task-level numRecordsOut counter if this operator
+    // is at the end of the operator chain.
+    //
+    // Return null if we should not use the numRecordsOut counter to track the records emitted
+    // by this operator.
+    private Counter getOperatorRecordsOutCounter(
```

Comment:
Better to annotate the method with `@Nullable`.

Reply:
Thanks for the suggestion. It is fixed now.